ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -153,6 +165,13 @@ a database ( directory )
           partitionsPostProcessing(iterator, scope, loadTaskTracker, tableTracker,
               partitionsTracker);
           partitionsTracker.debugLog("partitions");
+
+          if (!loadTaskTracker.hasReplicationState()) {
+            // Add ReplStateLogTask only if no pending table load tasks left for next cycle
+            ImportTableDesc tableDesc = loadPartitions.tableDesc();
+            loadTaskTracker.updateTaskCount(createTableReplLogTask(partitionsTracker.tasks(),
```

Comment:
if the task is added as dependency of the child table/partition/function tasks then you dont need to do anything as the DAG for those objects will increment the count traversing the whole DAG.